### PR TITLE
Added DNF package management implementation

### PR DESCRIPTION
**Pull request by schaefi**
DNF is the package manager used on Fedora systems and will also most likely replace yum for the next generation of RHEL. Along with the implementation here an example JeOS image description for Fedora can be found on: 

* https://github.com/SUSE/kiwi-descriptions

### CODE REVIEW

File: kiwi/repository/dnf.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+    def use_default_location(self):
+        """
+        Setup dnf repository operations to store all data
+        in the default places
+        """
+        self.shared_dnf_dir['reposd-dir'] = \
+            self.root_dir + '/etc/yum/repos.d'
+        self.shared_dnf_dir['cache-dir'] = \
+            self.root_dir + '/var/cache/yum'
```
Comment (Conan-Kudo) [inline kiwi/repository/dnf.py]: Default cache dir is `/var/cache/dnf`

File: kiwi/repository/dnf.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        self.shared_dnf_dir['pluginconf-dir'] = \
+            self.root_dir + '/etc/yum/pluginconf.d'
```
Comment (Conan-Kudo) [inline kiwi/repository/dnf.py]: Default plugin config dir is `/etc/dnf/plugins`

File: kiwi/repository/dnf.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        :param list custom_args: dnf arguments
+        """
+        self.custom_args = custom_args
+        if not custom_args:
+            self.custom_args = []
+
+        # extract custom arguments not used in dnf call
+        if 'exclude_docs' in self.custom_args:
+            self.custom_args.remove('exclude_docs')
+            log.warning('rpm-excludedocs not supported for dnf: ignoring')
```
Comment (Conan-Kudo) [inline kiwi/repository/dnf.py]: You can do this by doing `--setopt='tsflags=nodocs'`. There is a [pending PR for adding `--nodocs`](https://github.com/rpm-software-management/dnf/pull/748).

Comment (schaefi) [inline kiwi/repository/dnf.py]: Thanks, can this option also be part of the dnf config file, e.g

```
[main]
tsflags=nodocs
```

or would that not work ?

Comment (schaefi) [inline kiwi/repository/dnf.py]: Looking at https://github.com/fedora-cloud/docker-brew-fedora/issues/20 this should work for dnf and yum.

Comment (Conan-Kudo) [inline kiwi/repository/dnf.py]: That will work too!

**Comment by schaefi** [comment]
Thanks for the review 😄

**Pull request merged**